Test run on #330 caught a race condition

Need to investigate this race condition

https://github.com/abetterinternet/prio-server/pull/330/checks?check_run_id=1818399379

Relevant part of the log:

```
2021-02-02T22:46:22.2387388Z ==================
2021-02-02T22:46:22.2388417Z WARNING: DATA RACE
2021-02-02T22:46:22.2389057Z Read at 0x00c0000019c3 by goroutine 15:
2021-02-02T22:46:22.2389680Z   testing.(*common).logDepth()
2021-02-02T22:46:22.2390584Z       /opt/hostedtoolcache/go/1.15.7/x64/src/testing/testing.go:750 +0x164
2021-02-02T22:46:22.2391295Z   testing.(*common).log()
2021-02-02T22:46:22.2392058Z       /opt/hostedtoolcache/go/1.15.7/x64/src/testing/testing.go:737 +0x77
2021-02-02T22:46:22.2392761Z   testing.(*common).Log()
2021-02-02T22:46:22.2393514Z       /opt/hostedtoolcache/go/1.15.7/x64/src/testing/testing.go:776 +0x1d
2021-02-02T22:46:22.2395605Z   github.com/letsencrypt/prio-server/workflow-manager/limiter.TestLimiterBehavior.func2.1.1()
2021-02-02T22:46:22.2397500Z       /home/runner/work/prio-server/prio-server/workflow-manager/limiter/limiter_test.go:64 +0x136
2021-02-02T22:46:22.2398184Z 
2021-02-02T22:46:22.2398774Z Previous write at 0x00c0000019c3 by main goroutine:
2021-02-02T22:46:22.2399669Z   testing.tRunner.func1()
2021-02-02T22:46:22.2400519Z       /opt/hostedtoolcache/go/1.15.7/x64/src/testing/testing.go:1109 +0x484
2021-02-02T22:46:22.2401233Z   testing.tRunner()
2021-02-02T22:46:22.2402009Z       /opt/hostedtoolcache/go/1.15.7/x64/src/testing/testing.go:1127 +0x22a
2021-02-02T22:46:22.2402716Z   testing.runTests()
2021-02-02T22:46:22.2403513Z       /opt/hostedtoolcache/go/1.15.7/x64/src/testing/testing.go:1437 +0x612
2021-02-02T22:46:22.2404172Z   testing.(*M).Run()
2021-02-02T22:46:22.2404917Z       /opt/hostedtoolcache/go/1.15.7/x64/src/testing/testing.go:1345 +0x3b3
2021-02-02T22:46:22.2405551Z   main.main()
2021-02-02T22:46:22.2406031Z       _testmain.go:93 +0x356
2021-02-02T22:46:22.2406335Z 
2021-02-02T22:46:22.2406832Z Goroutine 15 (running) created at:
2021-02-02T22:46:22.2408358Z   github.com/letsencrypt/prio-server/workflow-manager/limiter.TestLimiterBehavior.func2.1()
2021-02-02T22:46:22.2410189Z       /home/runner/work/prio-server/prio-server/workflow-manager/limiter/limiter_test.go:58 +0x71
2021-02-02T22:46:22.2411662Z   github.com/letsencrypt/prio-server/workflow-manager/limiter.(*Limiter).Execute()
2021-02-02T22:46:22.2413152Z       /home/runner/work/prio-server/prio-server/workflow-manager/limiter/limiter.go:51 +0xd3
2021-02-02T22:46:22.2414882Z   github.com/letsencrypt/prio-server/workflow-manager/limiter.TestLimiterBehavior.func2()
2021-02-02T22:46:22.2416661Z       /home/runner/work/prio-server/prio-server/workflow-manager/limiter/limiter_test.go:57 +0x74
2021-02-02T22:46:22.2417439Z ==================
2021-02-02T22:46:22.2418188Z panic: Log in goroutine after TestLimiterBehavior has completed
2021-02-02T22:46:22.2418738Z 
2021-02-02T22:46:22.2419181Z goroutine 13 [running]:
2021-02-02T22:46:22.2419870Z testing.(*common).logDepth(0xc000122000, 0xc00001e1c0, 0x10, 0x3)
2021-02-02T22:46:22.2420800Z 	/opt/hostedtoolcache/go/1.15.7/x64/src/testing/testing.go:755 +0x7ee
2021-02-02T22:46:22.2421599Z testing.(*common).log(...)
2021-02-02T22:46:22.2422357Z 	/opt/hostedtoolcache/go/1.15.7/x64/src/testing/testing.go:737
2021-02-02T22:46:22.2423148Z testing.(*common).Log(0xc000122000, 0xc000039790, 0x1, 0x1)
2021-02-02T22:46:22.2423960Z 	/opt/hostedtoolcache/go/1.15.7/x64/src/testing/testing.go:776 +0x78
2021-02-02T22:46:22.2425935Z github.com/letsencrypt/prio-server/workflow-manager/limiter.TestLimiterBehavior.func2.1.1(0xc000010040, 0xc00001e1a0, 0xc00006a480, 0xc000122000)
2021-02-02T22:46:22.2427901Z 	/home/runner/work/prio-server/prio-server/workflow-manager/limiter/limiter_test.go:64 +0x137
2021-02-02T22:46:22.2429871Z created by github.com/letsencrypt/prio-server/workflow-manager/limiter.TestLimiterBehavior.func2.1
2021-02-02T22:46:22.2431688Z 	/home/runner/work/prio-server/prio-server/workflow-manager/limiter/limiter_test.go:58 +0x72
2021-02-02T22:46:22.2433087Z FAIL	github.com/letsencrypt/prio-server/workflow-manager/limiter	0.026s
2021-02-02T22:46:22.2434577Z ok  	github.com/letsencrypt/prio-server/workflow-manager/monitor	0.028s	coverage: 100.0% of statements
2021-02-02T22:46:22.2436029Z ?   	github.com/letsencrypt/prio-server/workflow-manager/task	[no test files]
2021-02-02T22:46:22.2437446Z ?   	github.com/letsencrypt/prio-server/workflow-manager/tokenfetcher	[no test files]
2021-02-02T22:46:22.2438874Z ?   	github.com/letsencrypt/prio-server/workflow-manager/utils	[no test files]
2021-02-02T22:46:22.2439586Z FAIL
2021-02-02T22:46:22.8588708Z ##[error]Process completed with exit code 1.
2021-02-02T22:46:22.8805152Z Post job cleanup.
```

--- a/workflow-manager/limiter/limiter_test.go
+++ b/workflow-manager/limiter/limiter_test.go
@@ -1,6 +1,7 @@
 package limiter
 
 import (
+	"sync"
 	"testing"
 	"time"
 )
@@ -25,11 +26,15 @@ func TestTicketDoubleDone(t *testing.T) {
 }
 
 func TestLimiterBehavior(t *testing.T) {
+	wg := sync.WaitGroup{}
+	// We have 3 goroutines we're spinning up
+	wg.Add(3)
 	firstJobChannel := make(chan struct{})
 
 	l := New(1)
 	l.Execute(func(ticket *Ticket) {
 		go func() {
+			defer wg.Done()
 			// Indicate this job is running and thus occupying the Limiter's
 			// single token, then block until instructed to terminate
 			firstJobChannel <- struct{}{}
@@ -56,6 +61,7 @@ func TestLimiterBehavior(t *testing.T) {
 	go func() {
 		l.Execute(func(ticket *Ticket) {
 			go func() {
+				defer wg.Done()
 				defer l.Done(ticket)
 				// Indicate that this job has started. Send a string to make the
 				// switch below more readable
@@ -69,6 +75,7 @@ func TestLimiterBehavior(t *testing.T) {
 	go func() {
 		l.Execute(func(ticket *Ticket) {
 			go func() {
+				defer wg.Done()
 				defer l.Done(ticket)
 				// Indicate that this job has started
 				messageChannel <- "third job"
@@ -98,6 +105,7 @@ func TestLimiterBehavior(t *testing.T) {
 				secondJobRan = true
 				if thirdJobRan {
 					t.Log("second and third jobs ran -- success")
+					wg.Wait()
 					return
 				}
 			case "third job":
@@ -107,6 +115,7 @@ func TestLimiterBehavior(t *testing.T) {
 				thirdJobRan = true
 				if secondJobRan {
 					t.Log("second and third jobs ran -- success")
+					wg.Wait()
 					return
 				}
 			}

